packages/adapter/adapter-node/src/response.ts

```diff
@@ -129,9 +130,17 @@ function writeHead(fetchResponse: Response, nodeResponse: ServerResponse) {
 	}
 }
 
+type GenericResponse = {
+	write(chunk: Uint8Array): boolean;
+	once(event: "drain", listener: () => void): void;
+	once(event: "error", listener: (err: unknown) => void): void;
+	off(event: "drain", listener: () => void): void;
+	off(event: "error", listener: (err: unknown) => void): void;
+};
+
 async function writeAndAwait(
 	chunk: Uint8Array,
-	res: ServerResponse,
+	res: GenericResponse,
```

Comment:
Needed because `write` methods of `http.ServerResponse` and `http2.Http2ServerResponse` are not compatible according to the type-checker, even though they are when called without a callback.